packages/lit-html/src/lit-html.ts

```diff
@@ -794,20 +812,24 @@ class TemplateInstance {
     return fragment;
   }
 
-  _update(values: Array<unknown>) {
+  _update(values: Array<unknown>): boolean {
```

Comment:
Since most of these objects end up having state flags anyway (`_hasDisconnect`), could we just read that flag rather than carefully thread a seemingly arbitrary return value up through these functions? I think that's likely less code overall.

Reply:
It compiles to a hair more code to read state flags rather than return a value. 🤷‍♂️

On top of that, in NodePart setValue, we still have to return a boolean from `_commitTemplateResult` and `_commitIterable`, since we can't reset the flag to false and let those set the flag to true because those commit functions need to clear based on the previous state of the flag. I'm inclined to leave it as-is for now, for these reasons, plus it's easier to grok what's going on imo.

If we need to pass more info up the tree, we have options to change it.